tasks/eva_3417/src/main/groovy/eva3417/RemediateLowerCaseRefAlt.groovy

```diff
@@ -0,0 +1,270 @@
+if (!options) {
+    cli.usage()
+    System.exit(1)
+}
+
+// call to remediate lowercase nucleotide
+new RemediateLowerCaseNucleotide(options.prodPropertiesFile, options.hashcollisionFile).remediate()
+
+class RemediateLowerCaseNucleotide {
+    private String hashCollisionFilePath
+    private EVADatabaseEnvironment dbEnv
+    private Function<ISubmittedVariant, String> hashingFunction
+
+    static def logger = LoggerFactory.getLogger(Application.class)
+
+    RemediateLowerCaseNucleotide(prodPropertiesFile, hashcollisionFile) {
+        this.dbEnv = createFromSpringContext(prodPropertiesFile, Application.class)
+        this.hashCollisionFilePath = hashcollisionFile
+        this.hashingFunction = new SubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction())
+    }
+
+    List<SubmittedVariantEntity> getImpactedSVEList(List<SubmittedVariantEntity> sveList) {
+        List<SubmittedVariantEntity> impactedSVEList = new ArrayList<>()
+        for (SubmittedVariantEntity sve : sveList) {
+            String ref = sve.getReferenceAllele()
+            String alt = sve.getAlternateAllele()
+            if (ref != ref.toUpperCase() || alt != alt.toUpperCase()) {
+                impactedSVEList.add(sve)
+            }
+        }
+
+        return impactedSVEList
+    }
+
+
+    String getSVENewHash(SubmittedVariantEntity sve) {
+        SubmittedVariant temp = sve.getModel()
+        temp.setReferenceAllele(sve.getReferenceAllele().toUpperCase())
+        temp.setAlternateAllele(sve.getAlternateAllele().toUpperCase())
+        return hashingFunction.apply(temp)
+    }
+
+    List<String> getExistingHashList(Collection<String> newHashList) {
+        List<SubmittedVariantEntity> existingHashSVEList = dbEnv.mongoTemplate
+                .find(query(where("_id").in(newHashList)), sveClass)
+
+        return existingHashSVEList.stream()
+                .map(sve -> sve.getHashedMessage())
+                .collect(Collectors.toList())
+    }
+
+    void insertSVEWithNewHash(List<SubmittedVariantEntity> sveList, Map<String, String> mapOldHashNewHash) {
+        List<SubmittedVariantEntity> sveWithNewHashList = new ArrayList<>()
+        for (SubmittedVariantEntity sve : sveList) {
+            // create a new SVE with updated ref, alt and hash
+            SubmittedVariant sveModel = sve.getModel()
+            sveModel.setReferenceAllele(sve.getReferenceAllele().toUpperCase())
+            sveModel.setAlternateAllele(sve.getAlternateAllele().toUpperCase())
+            SubmittedVariantEntity submittedVariantEntity = new SubmittedVariantEntity(sve.getAccession(),
+                    mapOldHashNewHash.get(sve.getHashedMessage()), sveModel, sve.getVersion())
+
+            sveWithNewHashList.add(submittedVariantEntity)
+        }
+
+        // insert updated SVEs into db
+        dbEnv.mongoTemplate.insert(sveWithNewHashList, sveClass)
+    }
+
+    void deprecateSVEWithOldHash(List<SubmittedVariantEntity> sveList) {
```

Comment:
I don't think we should deprecate these SSids.
I now understand deprecation as a statement about the associated id (SS or RS) and adding a deprecated operation says that this ssid is deprecated. That is not the case here.
I think what we meant is "this document has been updated" for which we have the UPDATED operation. 
If we do not want to overload the UPDATED operation (because it is associated with MERGED) we should create a new operation.

Reply:
removed deprecate and changed to update operation